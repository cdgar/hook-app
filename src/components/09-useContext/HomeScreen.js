import React, { useContext } from 'react'
import User from './User';
import UserContext from './UserContext'

const HomeScreen = () => {
    const {user} = useContext(UserContext);
    return (
        <div>
            <h1>HomeScreen</h1>
            <hr/>
            <User user={user}/>
        </div>
    )
}

export default HomeScreen
