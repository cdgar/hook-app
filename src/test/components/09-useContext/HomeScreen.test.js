import { mount } from 'enzyme';
import React from 'react';
import HomeScreen from '../../../components/09-useContext/HomeScreen';
import UserContext from '../../../components/09-useContext/UserContext';
import { userMock } from '../../fixtures/userMock';

describe('HomeScreen', () => {
    const wrapper = mount(
        <UserContext.Provider value={{user: userMock}}>
            <HomeScreen/>
        </UserContext.Provider>
    );

    test('should show HomeScreen', () => {
        expect(wrapper).toMatchSnapshot();
    })    
})