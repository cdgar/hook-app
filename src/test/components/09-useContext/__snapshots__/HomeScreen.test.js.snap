// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should show HomeScreen 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "nameXD",
   "email": "email@email.com"
}
  </pre>
</div>
`;
